src/Razor/src/Microsoft.AspNetCore.Razor.ExternalAccess.RoslynWorkspace/RazorWorkspaceListener.cs

```diff
@@ -160,17 +164,17 @@ protected virtual Task SerializeProjectAsync(ProjectId projectId, CancellationTo
             return Task.CompletedTask;
         }
 
+        _logger.LogTrace("{projectId} writing json file", projectId);
         return RazorProjectJsonSerializer.SerializeAsync(project, _projectRazorJsonFileName, ct);
     }
 
     public void Dispose()
     {
-        lock (_gate)
+        var queues = _workQueues;
+        _workQueues.Clear();
```

Comment:
Is this a typo? Clear will just return an empty immutable dictionary, but that value isn't captured here. Did you mean something like this instead?

```suggestion
        var queues = Interlocked.Exchange(ref _workQueues, ImmutableDictionary<ProjectId, TaskDelayScheduler>.Empty);
```

Reply:
🤦‍♂️ thank you, that is exactly what I meant.